src/main/java/net/minecraftforge/world/WorldCapabilityData.java

```diff
@@ -0,0 +1,63 @@
+package net.minecraftforge.world;
+
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.world.WorldProvider;
+import net.minecraft.world.WorldSavedData;
+import net.minecraftforge.common.capabilities.CapabilityDispatcher;
+import net.minecraftforge.common.util.INBTSerializable;
+
+public class WorldCapabilityData extends WorldSavedData
+{
+    private INBTSerializable<NBTTagCompound> serializable;
+    private NBTTagCompound data = null;
+    private int dimension;
+
+    public WorldCapabilityData(String name)
+    {
+        super(name);
+    }
+
+    public WorldCapabilityData(WorldProvider provider, INBTSerializable<NBTTagCompound> serializable)
+    {
+        super(fileNameForProvider(provider));
+        this.serializable = serializable;
+        this.dimension = provider.getDimension();
+    }
+
+	@Override
+    public void readFromNBT(NBTTagCompound nbt)
+    {
+	    if(nbt.hasKey("ForgeCaps"))
```

Comment:
Don't worlds already have a `Forge` compound tag? If so, why not add a `Capabilities` compound under the `Forge` compound?